pkg/controller/external_nofork.go

```diff
@@ -264,18 +278,53 @@ func (c *NoForkConnector) Connect(ctx context.Context, mg xpresource.Managed) (m
 	}
 
 	return &noForkExternal{
-		ts:             ts,
-		resourceSchema: c.config.TerraformResource,
-		config:         c.config,
-		params:         params,
-		rawConfig:      rawConfig,
-		logger:         logger,
-		metricRecorder: c.metricRecorder,
-		opTracker:      opTracker,
+		ts:                             ts,
+		resourceSchema:                 c.config.TerraformResource,
+		config:                         c.config,
+		params:                         params,
+		initProviderExclusiveParamKeys: initProviderExclusiveParams,
+		rawConfig:                      rawConfig,
+		logger:                         logger,
+		metricRecorder:                 c.metricRecorder,
+		opTracker:                      opTracker,
 	}, nil
 }
 
-func (n *noForkExternal) getResourceDataDiff(ctx context.Context, s *tf.InstanceState) (*tf.InstanceDiff, error) {
+func deleteInstanceDiffAttribute(instanceDiff *tf.InstanceDiff, paramKey string) error {
+	delete(instanceDiff.Attributes, paramKey)
+
+	keyComponents := strings.Split(paramKey, ".")
+	if len(keyComponents) < 2 {
+		return nil
+	}
+
+	keyComponents[len(keyComponents)-1] = "%"
```

Comment:
We should consider utilizing the resource Terraform schema to decide on the key of the length attribute.

Reply:
Because of time restrictions, we decided to do so later. We also added a to-do comment in the [next PR](https://github.com/ulucinar/upbound-upjet/pull/6).